### PR TITLE
Wikipedia bot

### PR DESCRIPTION
**Pull request by rxl881**

### CODE REVIEW

File: src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+		}, nil
+	}
+
+	// Truncate the extract text, if necessary
+	if len(extractText) > maxExtractLength {
+		extractText = extractText[:maxExtractLength] + "..."
+	}
+
+	// Add a link to the bottom of the extract
+	extractText += "\n" + "http://en.wikipedia.org/?curid=" + strconv.FormatInt(searchResultPage.PageID, 10)
```
Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia.go]:
```go
extractText += fmt.Sprintf(`\nhttp://en.wikipedia.org/?curid=%d`, searchResultPage.PageID)
```

may be clearer.

File: src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			StatusCode: 200,
+			Body:       ioutil.NopCloser(bytes.NewBuffer(b)),
+		}, nil
+	})
+	// clobber the Wikipedia service http client instance
+	httpClient = &http.Client{Transport: wikipediaTrans}
+
+	// Create the Wikipedia service
+	srv, err := types.CreateService("id", ServiceType, "@wikipediabot:hyrule", []byte(
+		`{"api_key":"`+apiKey+`"}`,
```
Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia_test.go]:
```go
fmt.Sprintf(`{"api_key":"%s"}`, apiKey)
```

**Comment by Kegsay** [comment]
Travis errors.

**Comment by rxl881** [comment]
@Kegsay - code should be a lot cleaner now (incorporating tips from previous pull request). Can you PTAFL?)

File: src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+const maxExtractLength = 1024 // Max length of extract string in bytes
+
+var httpClient = &http.Client{}
+
+type wikipediaSearchResults struct {
+	Query wikipediaQuery `json:"query"`
+}
+
+type wikipediaQuery struct {
+	Pages map[string]wikipediaPage `json:"pages"`
```
Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia.go]: It'd be good to comment what the keys represent here.

**Comment by Kegsay** [comment]
Other than the comment, LGTM!

File: src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia.go [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+var httpClient = &http.Client{}
+
+// Search results (returned by search query)
+type wikipediaSearchResults struct {
+	Query wikipediaQuery `json:"query"`
+}
+
+// Wikipeda pages returned in search results
+type wikipediaQuery struct {
+	Pages map[string]wikipediaPage `json:"pages"`
```
Comment (Kegsay) [inline src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia.go]: This is a map. What do the keys represent? Please comment.

Comment (rxl881) [inline src/github.com/matrix-org/go-neb/services/wikipedia/wikipedia.go]: Done. Thanks.

**Pull request merged**